pkg/restore/backup_extractor.go

```diff
@@ -87,12 +87,12 @@ func (e *backupExtractor) readBackup(tarRdr *tar.Reader) (string, error) {
 			if err != nil {
 				return "", err
 			}
-			defer file.Close()
 
 			if _, err := io.Copy(file, tarRdr); err != nil {
 				e.log.Infof("error copying: %v", err)
 				return "", err
 			}
+			file.Close()
```

Comment:
@asaf-erlich, this might leak resources. It might be safer to put the whole block regarding the file in a named or anonymous function and keep the defer statement. What do you think?

Reply:
I agree -- you could either wrap L85-95, like:
```
func() {
    // L85-95
}()
```

or you could create a helper func and move L85-95 into it. I'm happy with either approach.  If you use the anonymous func, please add a comment on why it's there.